### PR TITLE
Checkout: Show sublabel for DIFM extra pages

### PR DESCRIPTION
**Pull request by aneeshd16**
#### Proposed Changes

PT: pdh1Xd-Oi-p2
GH: 969-gh-Automattic/martech

The DIFM service currently includes 5 pages for the site build. We are in the process of shifting to a tiered pricing model for DIFM so that the user can purchase extra pages. If more than 5 pages are selected, this change adds a sublabel to the DIFM product on the checkout screen. 

The backend changes are in D85709-code. If tiered pricing is available, the `price_tier_maximum_units` field on the `product` object denotes the number of included pages (currently 5), and the `quantity` field denotes the number of pages selected by the user.

#### Testing Instructions

**Without the tiered pricing change**
* Go to `/start/do-it-for-me`, and complete the flow till you reach checkout. There should be no sublabel visible.
Screenshot 2022-08-23 at 6 13 23 PM

**With the tiered pricing change**
* Apply D85709-code
* Go to `/start/do-it-for-me?flags=difm/allow-extra-pages`. On the page picker step, select **less** than 5 pages. There should be no sublabel visible on checkout.
Screenshot 2022-08-23 at 6 13 23 PM

* Go to `/start/do-it-for-me?flags=difm/allow-extra-pages`. On the page picker step, select **exactly** 6 pages. The sublabel should be visible at the checkout screen and the text should be in the singular form (1 Extra Page)
Screenshot 2022-08-23 at 6 12 35 PM

* Go to `/start/do-it-for-me?flags=difm/allow-extra-pages`. On the page picker step, select **more** than 6 pages. The sublabel should be visible at the checkout screen and the text should be in the plural form (7 Extra Pages)
Screenshot 2022-08-23 at 5 51 47 PM

#### Pre-merge Checklist

- [x] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)? - Not required since the product is available only for EN.

Fixes Automattic/martech#969

### CODE REVIEW

File: packages/wpcom-checkout/test/checkout-line-items.tsx
Comment (sirbrillig) [inline packages/wpcom-checkout/test/checkout-line-items.tsx]: FWIW I think there's a project currently replacing `enzyme` with `testing-library`, eg:

- https://github.com/Automattic/wp-calypso/pull/65517
- https://github.com/Automattic/wp-calypso/pull/65234
- https://github.com/Automattic/wp-calypso/pull/65140
- https://github.com/Automattic/wp-calypso/pull/65019
- etc.

We might not want to introduce another test that needs to be migrated. Could this use `testing-library` instead?

**Review by sirbrillig — COMMENTED** [review]
I haven't run through the testing instructions (I leave that to your team), but the code changes look sensible to me. Nice job using the `isSmallestUnit` option in `formatCurrency`! I had one suggestion about the tests, which I've left inline.

File: packages/wpcom-checkout/test/checkout-line-items.tsx
Comment (aneeshd16) [reply packages/wpcom-checkout/test/checkout-line-items.tsx]: Thanks for the review @sirbrillig! And thanks for sharing examples about migrating to `testing-library`. (TIL!)
I have updated the PR and migrated the tests to `testing-library`.

**Review by ddc22 — CHANGES_REQUESTED** [review]
Hi @aneeshd16  I’ve been thinking about it and I think we should rebase this onto the extra pages super PR (#62848) that we have been maintaining. I am planning to change my PR(#66559) similarly. Having to do code level changes without which a change is not visible makes me a bit uneasy.

There is no value in directly merging this to trunk as well since it cannot be used before we merge that Super PR anyway.

**Comment by aneeshd16** [comment]
> Hi @aneeshd16 I’ve been thinking about it and I think we should rebase this onto the extra pages super PR (https://github.com/Automattic/wp-calypso/pull/62848) that we have been maintaining. I am planning to change my PR(https://github.com/Automattic/wp-calypso/pull/66559) similarly. Having to do code level changes without which a change is not visible makes me a bit uneasy.

As discussed, we will be hiding the new sublabel behind a feature flag. @jdc91 Can you please have a look at the new change?

File: packages/wpcom-checkout/src/checkout-line-items.tsx
Comment (ddc22) [inline packages/wpcom-checkout/src/checkout-line-items.tsx]: Do we need to check this flag here? I think we can remove it since what we already block the user in the previous steps and he has no way to select extra steps without that step.

**Review by ddc22 — APPROVED** [review]
Works as expected 🚢 
Minor commend added below.

File: packages/wpcom-checkout/src/checkout-line-items.tsx
Comment (aneeshd16) [reply packages/wpcom-checkout/src/checkout-line-items.tsx]: You're right, the flag is not required since the PR handles both cases (with and without tiered pricing).

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7493550

Thank you @aneeshd16 for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7493550).

**Pull request merged**
